### PR TITLE
Updates to Readme

### PR DESCRIPTION
**Pull request by dhruvbatra**
## Motivation and Context

Better formatting of installation instructions based on repeated questions in issues. 

## How Has This Been Tested

By following these instructions on my mac laptop (interactive testing) and SkyNet (non-interactive testing). 

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -193,42 +248,44 @@ To run the above benchmarks on your machine, see instructions in the [examples](
...
-## Developer installation and getting started
+We support transfering rendering results directly to a [PyTorch](https://pytorch.org/) tensor via CUDA-GL Interop.
+This feature is built by when Habitat-Sim is compiled with CUDA, i.e. built with `--with-cuda`.  To enable it, set the
+`gpu2gpu_transfer` flag of the sensor specification(s) to `True`
 
-1. Clone the repo.
-1. Install numpy in your python env of choice (e.g., `pip install numpy` or `conda install numpy`)
-1. Install dependencies in your python env of choice (e.g., `pip install -r requirements.txt`)
-1. If you are using a virtual/conda environment, make sure to use the same environment throughout the rest of the build
-1. Build using `./build.sh` or `./build.sh --headless` for headless systems (i.e. without an attached display) or if you need multi-gpu support
```
Comment (mathfac) [inline README.md]: Using `./build.sh` for development is important part.

File: README.md [hunk 2]
```diff
@@ -139,52 +134,112 @@ Previous simulation platforms that have operated on similar datasets typically p
...
+    Common
+   ```bash
+   # We require python>=3.6 and cmake>=3.10
+   conda create -n habitat python=3.6 cmake=3.14.0
+   conda activate habitat
+   pip install -r requirements.txt
+   ```
+
+    Linux (Tested with Ubuntu 18.04 with gcc 7.4.0)
+   ```bash
```
Comment (erikwijmans) [inline README.md]: Can we add a comment like `These are fairly ubiquitous packages and your system likely has them already`?  I don't want people to see something that requires `sudo` and stopping right there.

File: README.md [hunk 3]
```diff
@@ -139,52 +134,112 @@ Previous simulation platforms that have operated on similar datasets typically p
...
+   python setup.py install --bullet    # build habitat with bullet physics
+   ```
 
+   Note: some Linux distributions might require an additional `--user` flag to deal with permission issues. 
 
-**Note**: the build requires `cmake` version 3.10 or newer. You can install cmake through `conda install cmake` or download directly from [https://cmake.org/download/](https://cmake.org/download/)
 
+1. [Optional] For use with [Habitat-API](https://github.com/facebookresearch/habitat-api) and your own python code, add habitat-sim to your `PYTHONPATH`. For example modify your `.bashrc` (or `.bash_profile` in Mac OS X) file by adding the line:
+   ```bash
```
Comment (erikwijmans) [inline README.md]: Not needed for the `setup.py install` method

File: README.md (hunk 1 above)
Comment (erikwijmans) [reply README.md]: Seconded.  Due to silliness with c++ libraries and python,  `pytest` doesn't work with the `python setup.py install` build method.

File: README.md [hunk 4]
```diff
@@ -139,52 +134,112 @@ Previous simulation platforms that have operated on similar datasets typically p
...
-## Quick installation
+## Installation
+
+### Docker Image
+
+We provide a pre-built docker container for habitat-api and habitat-sim, refer to [habitat-docker-setup](https://github.com/facebookresearch/habitat-api#docker-setup).
+
+### From Source
+
+We highly recommend installing an [Anaconda](https://www.anaconda.com/distribution/#download-section) environment (note: python>=3.6 is required). Once you have Anaconda installed, here are the instructions.
```
Comment (erikwijmans) [inline README.md]:
```suggestion
We highly recommend installing an [miniconda](https://docs.conda.io/en/latest/miniconda.html) or [Anaconda](https://www.anaconda.com/distribution/#download-section) environment (note: python>=3.6 is required). Once you have Anaconda installed, here are the instructions.
```

Add pointer to miniconda -- it is much lighter weight to install and we don't need everything in pre-packaged anaconda.

File: README.md (hunk 1 above)
Comment (dhruvbatra) [reply README.md]: Agreed. I wasn't suggesting removing build.sh. I was only suggesting dropping this separation between "Quick Install" and "Developer Install". 

I'll add the information about "build.sh" back in the readme.

**Comment by dhruvbatra** [comment]
Incorporated changes.

File: README.md [hunk 5]
```diff
@@ -139,52 +134,117 @@ Previous simulation platforms that have operated on similar datasets typically p
...
+   ```bash
+   python setup.py install --bullet    # build habitat with bullet physics
+   ```
+
+   Note1: some Linux distributions might require an additional `--user` flag to deal with permission issues. 
+   
+   Note2: for active development in Habitat, you might find `./build.sh` instead of `python setup.py install` more useful. 
+
+
+1. [Optional; Needed only with `build.sh`] For use with [Habitat-API](https://github.com/facebookresearch/habitat-api) and your own python code, add habitat-sim to your `PYTHONPATH`. For example modify your `.bashrc` (or `.bash_profile` in Mac OS X) file by adding the line:
```
Comment (erikwijmans) [inline README.md]:
```suggestion
1. [Only if using `build.sh`] For use with [Habitat-API](https://github.com/facebookresearch/habitat-api) and your own python code, add habitat-sim to your `PYTHONPATH`. For example modify your `.bashrc` (or `.bash_profile` in Mac OS X) file by adding the line:
```

Doing that can actually cause things to break when using `setup.py install` as sometimes the python code and bindings will get out of sync without you realizing and/or the bindings won't get picked up at all.

**Pull request merged**